File: Numbers/mirror_distance_of_an_integer/TS/mirrorDistance.ts
function mirrorDistance(n: number): number {
    let number = n;
    let reverseNumber = 0;
    while(n > 0){
        reverseNumber = (reverseNumber * 10) + (n % 10)
        n = Math.floor(n / 10)
    }
    return Math.abs(number - reverseNumber)
};

File: Numbers/mirror_distance_of_an_integer/go/mirror_distance.go
package main

import (
	"fmt"
	"math"
	"strings"
)

func mirrorDistance(n int) int {
	reverse := 0
	for number := n; number > 0; {
		reverse = (reverse * 10) + (number % 10)
		number /= 10
	}
	return int(math.Abs(float64(reverse - n)))
}

func main() {
	test_cases := []int{
		25, 10, 7, 121, 100, 1,
	}

	for _, n := range test_cases {
		fmt.Printf("Input:	%v\n", n)
		fmt.Printf("Output:	%v\n", mirrorDistance((n)))
		fmt.Println(strings.Repeat("-", 35))
	}
}
